test/component.spec.ts

```diff
@@ -269,48 +272,40 @@ test("subraph component", function ({ components, stubComponents }) {
           })
 
           it("should retry the supplied amount of times", async () => {
-            const { fetch } = components
-
-            fetchMock.mockReset()
-            fetchMock = jest.spyOn(fetch, "fetch").mockImplementation()
+            fetchMock.mockReset().mockImplementation()
 
             try {
               await subgraph.query("query")
-            } catch (error) {
-              expect(fetchMock).toHaveBeenCalledTimes(retries + 1)
-            }
+            } catch (error) {}
+
+            expect(fetchMock).toHaveBeenCalledTimes(retries + 1)
           })
         })
       })
 
       describe("when the timeout is reached", () => {
-        const timeout = 500
         let subgraph: ISubgraphComponent
 
+        const errorMessage = "Query timed out"
+
         beforeEach(async () => {
-          const { config, fetch } = components
-          jest.spyOn(config, "getNumber").mockImplementation(async (name: string) => {
-            switch (name) {
-              case "SUBGRAPH_COMPONENT_QUERY_TIMEOUT":
-                return timeout
-              case "SUBGRAPH_COMPONENT_TIMEOUT_INCREMENT":
-                return 1
-              default:
-                return 0
-            }
-          })
+          const { fetch } = components
 
-          fetchMock = jest.spyOn(fetch, "fetch").mockImplementation(() => setTimeout(timeout + 1000))
+          let reject: Function
+          const fetchPromise = new Promise((_resolve, rej) => {
+            reject = rej
+          })
+          fetchMock = jest.spyOn(fetch, "fetch").mockImplementation(() => fetchPromise as any)
+          ;(setTimeout as jest.Mock).mockReset().mockImplementation(() => {
+            reject(new Error(errorMessage))
+            return Promise.resolve()
+          })
```

Comment:
If `setTimeout` uses under its hood the `setTimeout` API, you can use the `jest.useFakeTimers` API to run the timers and abort the fetch as it would do in a real life situation.

Reply:
I'm not sure if that'll work but i'll try!

Reply:
it unfortunately does not work. You think the current impl is good enough? I can investigate further